src/import/FSHImporter.ts

```diff
@@ -1760,22 +1787,32 @@ export class FSHImporter extends FSHVisitor {
     return pathRule;
   }
 
-  visitCodeInsertRule(ctx: pc.CodeInsertRuleContext): InsertRule {
+  visitCodeInsertRule(ctx: pc.CodeInsertRuleContext, keepSystem = false): InsertRule {
     const insertRule = new InsertRule('')
       .withLocation(this.extractStartStop(ctx))
       .withFile(this.currentFile);
     const localCodePath = ctx.CODE().map(code => {
-      return this.parseCodeLexeme(code.getText(), ctx).code;
+      const parsedCode = this.parseCodeLexeme(code.getText(), ctx);
+      if (keepSystem) {
+        return `${parsedCode.system ?? ''}#${parsedCode.code}`;
+      } else {
+        return parsedCode.code;
+      }
     });
     const fullCodePath = this.getArrayPathWithContext(localCodePath, ctx);
     insertRule.pathArray = fullCodePath;
     return this.applyRuleSetParams(ctx, insertRule);
   }
 
-  visitInsertRule(ctx: pc.InsertRuleContext): InsertRule {
-    const insertRule = new InsertRule(this.getPathWithContext(this.visitPath(ctx.path()), ctx))
+  visitInsertRule(ctx: pc.InsertRuleContext, withPathArray = false): InsertRule {
+    const localPath = this.visitPath(ctx.path());
+    const fullPathArray = this.getArrayPathWithContext(localPath === '' ? [] : [localPath], ctx);
```

Comment:
Is this array stuff only needed because of the undocumented syntax that allows introducing multiple codes from the same system in a single VS compose rule? If so, should we consider removing support for that syntax since it's not documented in the spec? (We'd want to check to see if people are using it before we removed it -- and check with Mark too).

Reply:
I'm storing information in `pathArray` because the path might be an actual path or a list of concepts, and that determination gets made elsewhere. So even with removing the syntax for multiple concepts in a single rule, tracking both is useful.

Alternatively, since ValueSets have the special condition of disallowing paths on their normal CaretValueRules, it might be possible to do this without needing to bother with `pathArray` here if I can be careful about always removing the path when it isn't a concept. Then, when exporting the ValueSet, any CaretValueRule with a path is assumed to be referring to something within a compose element.